### DISCUSSION TITLE
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!

### DISCUSSION THREAD

**Discussion by Sea0Wolf** [lllyasviel/Fooocus#2901] [Q&A]
I have a 9th generation I7, a 3070 and 32 ram, I have comflyUI installed on the same disk, but it won't let me generate in fooocus I get this error 

RuntimeError: All tensioners were expected to be on the same device, but they were found at least two devices, cuda:0 and cpu!

**Comment by mashb1t** [comment]
Please ensure you have sufficient swap set up as described in the [troubleshooting guide](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md#system-swap).

**Reply by mmakar16** [reply]
Have a same problem. I did everything according to your instructions. The default settings were the same. There is about 1 TB of free disk space. But it still gives this error

**Reply by mmakar16** [reply]
D:\FOOCUS\Fooocus_win64_2-5-0>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Fast-forward merge
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.5
Error checking version for opencv-contrib-python-headless: No package metadata was found for opencv-contrib-python-headless
Installing requirements
[Cleanup] Attempting to delete content of temp dir C:\Users\User\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Downloading: "https://huggingface.co/mashb1t/misc/resolve/main/xl-to-v1_interposer-v4.0.safetensors" to D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\vae_approx\xl-to-v1_interposer-v4.0.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 5.40M/5.40M [00:00<00:00, 6.24MB/s]
Downloading: "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/juggernautXL_v8Rundiffusion.safetensors" to D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors

100%|█████████████████████████████████████████████████████████████████████████████| 6.62G/6.62G [09:09<00:00, 12.9MB/s]
Total VRAM 8191 MB, total RAM 32678 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3050 : native
VAE dtype: torch.bfloat16
Using pytorch cross attention
Refiner unloaded.
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
model_type EPS
UNet ADM Dimension 2816
IMPORTANT: You are using gradio version 3.41.2, however version 4.29.0 is available, please upgrade.
--------
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection'}
left over keys: dict_keys(['cond_stage_model.clip_l.transformer.text_model.embeddings.position_ids'])
Base model loaded: D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors
VAE loaded: None
Request to load LoRAs [('sd_xl_offset_example-lora_1.0.safetensors', 0.1)] for model [D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors].
Loaded LoRA [D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\models\checkpoints\juggernautXL_v8Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\_utils.py:831: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  return self.fget.__get__(instance, owner)()
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
loading in lowvram mode 64.0
lowvram: loaded module regularly Embedding(50257, 768)
lowvram: loaded module regularly Embedding(1024, 768)
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly Conv1D()
lowvram: loaded module regularly LayerNorm((768,), eps=1e-05, elementwise_affine=True)
lowvram: loaded module regularly Linear(in_features=768, out_features=50257, bias=False)
[Fooocus Model Management] Moving model(s) has taken 0.95 seconds
Started worker with PID 12752
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] CLIP Skip = 2
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] Seed = 8372004704515629838
[Parameters] CFG = 4
[Fooocus] Loading control models ...
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
Fooocus Expansion loaded by itself.
[Fooocus Model Management] Moving model(s) has taken 0.43 seconds
Traceback (most recent call last):
  File "D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\modules\async_worker.py", line 1471, in worker
    handler(task)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\modules\async_worker.py", line 1160, in handler
    tasks, use_expansion, loras, current_progress = process_prompt(async_task, async_task.prompt, async_task.negative_prompt,
  File "D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\modules\async_worker.py", line 738, in process_prompt
    expansion = pipeline.final_expansion(t['task_prompt'], t['task_seed'])
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\Fooocus\extras\expansion.py", line 120, in __call__
    features = self.model.generate(**tokenized_kwargs,
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\generation\utils.py", line 1914, in generate
    result = self._sample(
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\generation\utils.py", line 2651, in _sample
    outputs = self(
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 1421, in forward
    transformer_outputs = self.transformer(
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 1235, in forward
    outputs = block(
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 720, in forward
    attn_outputs = self.attn(
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 346, in forward
    attn_output, attn_weights = self._attn(query, key, value, attention_mask, head_mask)
  File "D:\FOOCUS\Fooocus_win64_2-5-0\python_embeded\lib\site-packages\transformers\models\gpt2\modeling_gpt2.py", line 217, in _attn
    attn_weights = torch.where(causal_mask, attn_weights.to(attn_weights.dtype), mask_value)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
Total time: 16.19 seconds
